aten/src/ATen/native/cuda/ScanKernels.cu

```diff
@@ -244,16 +244,16 @@ void cummin_helper_cuda(const Tensor& self, Tensor& values, Tensor& indices, int
  */
 template<typename scalar_t, class BinaryOp>
 __global__ void tensor_kernel_scan_outer_dim(scalar_t *tgt_, scalar_t *src_,
-                                              int num_orows, int num_irows, int row_size,
+                                              unsigned num_orows, unsigned num_irows, unsigned row_size,
```

Comment:
tensors can, in rare cases, have more rows than 2^32 (or have bigger row_size). It's ok to error out in this case, because using int64_t for indexing incurs perf penalty. But here it will just be a silent error, right?

Reply:
Oh. I didn't think of it. The performance was poor with signed int and when I checked with the previous implementation that was a difference.

Yes. Here the value will wrap around.

Is there any operation which errors out on getting row size above 2^32 so that I can check that for reference.

Reply:
Have added bounds check based on #38970 

Thanks @peterbell10